### PR TITLE
Minor semantic tokens cleanup

### PR DESCRIPTION
**Pull request by sebastiansturm**
this should fix bug #2920 (not the original issue, the one discussed further down in the issue thread), and removes one unneeded call to `lsp--semantic-tokens-initialize-buffer`.

### CODE REVIEW

File: lsp-semantic-tokens.el [hunk 1]
```diff
@@ -61,6 +61,15 @@ preferred over both full and ranged token requests."
   :group 'lsp-semantic-tokens
   :type 'boolean)
 
+(defcustom lsp-semantic-tokens-honor-refresh-requests nil
```
Comment (ericdallo) [inline lsp-semantic-tokens.el:64]: This frequency should be different form language server to language server so maybe it's a good idea to have it disabled or even in the future we allow/enable for servers we know do that "right"

**Review by ericdallo — APPROVED** [review]

**Comment by sebastiansturm** [comment]
strange that I'm getting CI failures on macos again; last time, simply rerunning the tests made those checks pass. Is there anything special about the macos CI environment?

**Comment by ericdallo** [comment]
@sebastiansturm AFAIK no, we noticed infra issues on the CI only, like `cask` not being able to find a dep on melpa or something like this, but nothing related with a specific job

**Comment by ericdallo** [comment]
I'd suggest re-run the jobs

**Pull request merged**